ReconBuilder.py

```diff
@@ -0,0 +1,323 @@
+# (topVertex, bottomVertex, edge1, edge2)
+# where topVertex is the top vertex of the edge, bottomVertex is the bottom vertex of the edge
+# and edge1 and edge2 are the two children edges or None (in the case of an edge terminating at a leaf)
+# Example host tree
+host_tree = { 'hTop': ('Top', 'm0', ('m0', 'm1'), ('m0', 'm2')),
+ ('m0', 'm1'): ('m0', 'm1', None, None),
+ ('m0', 'm2'): ('m0', 'm2', ('m2', 'm3'), ('m2', 'm4')),
+ ('m2', 'm3'): ('m2', 'm3', None, None),
+ ('m2', 'm4'): ('m2', 'm4', None, None)}
+
+# Example parasite tree
+parasite_tree = { 'pTop': ('Top', 'n0', ('n0', 'n1'), ('n0', 'n2')),
+  ('n0', 'n1'): ('n0', 'n1', None, None),
+  ('n0', 'n2'): ('n0', 'n2', ('n2', 'n3'), ('n2', 'n4')),
+  ('n2', 'n3'): ('n2', 'n3', None, None),
+  ('n2', 'n4'): ('n2', 'n4', None, None)}
+
+# Example host root:  
+host_root = 'm0'
+
+# Example parasite root: 
+parasite_root = 'n0'
+
+# Example tip mapping:  
+tip_mapping= {'n1': 'm4', 'n3': 'm4', 'n4': 'm4'}
+
+# Example reconciliation:
+reconciliation = { ('n0', 'm4'): [('D', ('n1', 'm4'), ('n2', 'm4'))],
+('n1', 'm4'): [('C', (None, None), (None, None))],
+('n2', 'm4'): [('D', ('n3', 'm4'), ('n4', 'm4'))],
+('n3', 'm4'): [('C', (None, None), (None, None))],
+('n4', 'm4'): [('C', (None, None), (None, None))]}
+
+# Example frequency dictionary
+# Provides frequency information for all non-contemporaneous associations
+# ROSE:  YOU DON'T NEED TO WORRY ABOUT THIS
+frequencies = { ('D', ('n1', 'm4'), ('n2', 'm4')): 0.5,
+                ('D', ('n3', 'm4'), ('n4', 'm4')): 0.75 }
+
+def build_formatted_tree(tree):
+    """
+    :param tree:  a tree dictionary
+    :return: A temporal graph that contains all the temporal relations implied by
+             the tree. Each key is a node tuple of the form (name, type) where name
+             is a string representing the name of a parasite or host tree INTERNAL 
+             node and type is either TreeType.HOST or TreeType.PARASITE which are 
+             defined in Recon.py. The associated value is a list of node tuples that
+             are the children of this node tuple in the tree.
+    """
+    tree_type = None
+    if 'pTop' in tree:
+        tree_type = TreeType.PARASITE
+    else:
+        tree_type = TreeType.HOST
+
+    formatted_tree = {}
+    for edge_name in tree:
+        edge_four_tuple = tree[edge_name]
+        # the temporal graph does not contain leaves as keys
+        if _is_leaf_edge(edge_four_tuple):
+            continue
+        # the temporal graph contains internal node tuples as keys,
+        # and their children nodes tuples as values
+        node_name = _bottom_node(edge_four_tuple)
+        left_child_name = edge_four_tuple[2][1]
+        right_child_name = edge_four_tuple[3][1]
+        formatted_tree[(node_name, tree_type)] = [(left_child_name, tree_type), \
+                                               (right_child_name, tree_type)]
+    return formatted_tree
+
+def create_parent_dict(H, P):
```

Comment:
I think it would be nice if we create a class for this. What do you think?

Reply:
I think this would be a possibility after we refactor the code to use classes. To be consistent with the style of other parts of the code, I will keep the dictionary structure in this commit.